src/main/webapp/resources/js/components/samples/components/SampleFileContenate.tsx

```diff
@@ -161,18 +175,27 @@ export function SampleFileConcatenate({
           className="t-concatenate-confirm-modal"
           onCancel={() => {
             dispatch(resetConcatenateSelected());
+            setConcatenationError("");
+            setConcatenateButtonDisabled(true);
             setVisible(false);
           }}
           visible={visible}
           onOk={concatenateFiles}
           okText={i18n("SampleFilesConcatenate.okText")}
           cancelText={i18n("SampleFilesConcatenate.cancelText")}
-          okButtonProps={{ className: "t-concatenate-confirm" }}
+          okButtonProps={{
+            className: "t-concatenate-confirm",
+            disabled: concatenateButtonDisabled,
+            icon: concatenating && <LoadingOutlined />,
```

Comment:
There is a loading prop on Button that will automatically add the icon and set the disabled state

Reply:
Updated in 2c94eb7d49c94c036f27427d7d02746f56a089d3